test/MLJ/unit.jl

```diff
@@ -186,6 +195,104 @@ end
 
     end
 
+    # an example of a generic kwarg constructor which has been previously unsupported in TreeParzen
+    testGenericKwargModel = DummyGenericKwargModel(x=5.)
+    # example single and nested/layered spaces
+    entry_A = :x => HP.Uniform(:x, 6., 10.)
+    entry_B = :y => HP.Uniform(:y, -3., 3.)
+    single_space = Dict(entry_A)
+    joint_space = Dict(:joint => TreeParzen.HP.Choice(:joint, [Dict(entry_A), Dict(entry_B)]))
+    nested_space = Dict(
+        :z => HP.Choice(
+            :z,
+            [
+                Dict(:z => true),
+                Dict(
+                    :z => true,
+                    entry_A
+                ),
+                Dict(
+                    :z => false,
+                    entry_B
+                )
+            ]
+        )
+    )
+
+    @testset "no suggestions with a generic kwarg model constructor example --single space" begin
+
+        suggestions = Dict{Symbol}[]
+        tuning, state = setup(;n_startup=3,space=single_space, suggest=suggestions)
+
+        output, _ = MLJTuning.models(tuning, testGenericKwargModel, nothing, state, 0, 0)
+
+        @test output isa Vector{<:Tuple{Any, TreeParzen.Trials.Trial}}
+        @test length(output) == 3
+        param_x_values = getindex.(getproperty.(last.(output), :hyperparams), :x)
+        # given a defined space with uniform distribution between 6. and 10.
+        # the tests below checks that the param x from the dummy model was updated
+        # in each trial within that range
+        @test minimum(param_x_values) > 6.
+        @test maximum(param_x_values) < 10.
+        # checking that default was updated although it is possible to have an odd case where
+        # all results are 5.
+        @test !all(in.(param_x_values, [(5., 5., 5.)]))
+
+    end
+
+    @testset "no suggestions with a generic kwarg model constructor example --list of Dicts" begin
+
+        suggestions = Dict{Symbol}[]
+        tuning, state = setup(;n_startup=3,space=joint_space, suggest=suggestions)
+
+        output, _ = MLJTuning.models(tuning, testGenericKwargModel, nothing, state, 0, 0)
+
+        @test output isa Vector{<:Tuple{Any, TreeParzen.Trials.Trial}}
+        @test length(output) == 3
+        joint_values = getindex.(getproperty.(last.(output), :hyperparams), :joint)
+        # checks that parameters x and y were updated within their uniform distributions
+        # and that they differ from their defaults
+        for d in joint_values
+            if haskey(d, :x)
+                @test all(6. .<= d[:x] .<= 10.)
```

Comment:
right, so I removed broadcasting as it's not needed because each `d[:x]` (if `:x` parameter was chosen) is a single value not a vector for the `:x` parameter and the `joint_values` is a vector of dictionaries with the final choices given the setup.

Reply:
can you remove "all"?